desktop/apps/auction/client/analytics.js

```diff
@@ -0,0 +1,57 @@
+import * as actions from './actions'
+import analyticsHooks from '../../../lib/analytics_hooks.coffee'
+import { contains, isEqual } from 'underscore'
+
+const analyticsMiddleware = store => next => action => {
+  const result = next(action)
+  const nextState = store.getState()
+
+  // track certain types of actions
+  switch(action.type) {
+    case actions.UPDATE_MEDIUM_ID: {
+      trackParamChange({ medium: action.payload.mediumId }, nextState)
+      return result
+    }
+    case actions.UPDATE_ARTIST_ID: {
+      trackParamChange({ artist: action.payload.artistId }, nextState)
+      return result
+    }
+    case actions.UPDATE_SORT: {
+      trackParamChange({ sort: action.payload.sort }, nextState)
+      return result
+    }
+    case actions.UPDATE_ESTIMATE_RANGE: {
+      trackParamChange({
+        price: nextState.auctionArtworks.filterParams.estimate_range
+      }, nextState)
+      return result
+    }
+    default: return result
+  }
+}
+
+function trackableArtistIds(changed, filterParams) {
```

Comment:
This method is necessary to coerce our internal filter param definition into something more useful for the analytics team. (i.e. internally, we filter by "artists you follow" by setting `include_followed_artists: true` and `artist_ids: []`, but the analytics team would rather have `artists: ['artists-you-follow']`).